extensions/blocks/podcast-player/podcast-player.js

```diff
@@ -0,0 +1,372 @@
+	 * @returns {object} Track object.
+	 */
+	getTrack( track ) {
+		return this.tracks[ track ];
+	}
+
+	/**
+	 * Error handler for audio.
+	 * @private
+	 */
+	handleError() {
+		this.setPlayerState( STATE_ERROR );
+		if ( this.currentTrack > -1 ) {
+			this.setTrackState( this.currentTrack, true, true );
+		}
+	}
+
+	/**
+	 * Play handler for audio.
+	 * @private
+	 */
+	handlePlay() {
+		this.setPlayerState( STATE_PLAYING );
+		if ( this.currentTrack === -1 ) {
+			this.currentTrack = 0;
+			this.setTrackState( this.currentTrack, true );
+		}
+	}
+
+	/**
+	 * Pause handler for audio.
+	 * @private
+	 */
+	handlePause() {
+		// Ignore pauses if we are showing an error.
+		if ( this.playerState === STATE_ERROR ) {
+			return;
+		}
+
+		this.setPlayerState( STATE_PAUSED );
+	}
+
+	/**
+	 * Play current audio.
+	 * @public
+	 */
+	play() {
+		// Ignoring exceptions as they are handled globally from the audio element.
+		this.audio.play().catch( noop );
+	}
+
+	/**
+	 * Pause current audio.
+	 * @public
+	 */
+	pause() {
+		this.audio.pause();
+	}
+
+	/**
+	 * Stop current audio, clean selected track.
+	 * @public
+	 */
+	stop() {
+		this.pause();
+		this.setTrackState( this.currentTrack, false );
+		this.currentTrack = -1;
+	}
+
+	/**
+	 * Toggle playing state.
+	 * @public
+	 */
+	togglePlayPause() {
+		if ( this.audio.paused ) {
+			this.play();
+		} else {
+			this.pause();
+		}
+	}
+
+	/**
+	 * Select a track and play/pause, as needed.
+	 * @public
+	 * @param {number} track The track number
+	 */
+	selectTrack( track ) {
+		// Current track already selected.
+		if ( this.currentTrack === track ) {
+			this.togglePlayPause();
+			return;
+		}
+
+		// Something else is playing.
+		if ( this.currentTrack !== -1 ) {
+			this.stop();
+		}
+
+		// Load a new track.
+		this.loadAndPlay( track );
+	}
+
+	/**
+	 * Set a player state classname.
+	 * @private
+	 * @param {string} state Player state.
+	 */
+	setPlayerState( state ) {
+		this.playerState = state;
+		this.block.classList.remove( STATE_ERROR, STATE_PAUSED, STATE_PLAYING );
+		this.block.classList.add( state );
+	}
+
+	/**
+	 * Load audio from the track, start playing.
+	 * @private
+	 * @param {number} track The track number
+	 */
+	loadAndPlay( track ) {
+		const trackData = this.getTrack( track );
+		if ( ! trackData ) {
+			return;
+		}
+		this.audio.src = trackData.src;
+		this.currentTrack = track;
+		this.play();
+		this.setTrackState( track, true );
+	}
+
+	/**
+	 * Visually mark track active or inactive.
+	 * @private
+	 * @param {number} track The track number
+	 * @param {boolean} isActive True if track is active
+	 * @param {boolean} isError True if track couldn't be played
+	 */
+	setTrackState( track, isActive, isError = false ) {
+		// Mark track element as active/inactive.
+		const el = this.trackElements[ track ];
+
+		// Active state classname.
+		if ( isActive ) {
+			el.classList.add( 'is-active' );
+		} else {
+			el.classList.remove( 'is-active' );
+		}
```

Comment:
This can be `el.classList.toggle( 'is-active', isActive )`.

Reply:
TIL about this but also learned it's a no-go in IE11 :( IE11 supports `toggle` but not the second param and I think we shouldn't shim it if the method exists. We did shim `closest` but that's 100% missing so there is no possibility it will break things

Reply:
and IE11 also doesn't support multiparam `add` and `remove`. I'll need to make more code a bit more ugly I guess.